File: angle-chan-server/src/main/java/com/ben/dektor/entities/Post.java
package com.ben.dektor.entities;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.sql.Date;
import java.util.List;

@Data
@Entity
public class Post {
    @Id
    private long number;
    private String name;
    @CreationTimestamp
    private Date utc;
    private String text;
    @OneToMany(cascade = CascadeType.ALL)
    private List<MediaContainer> media;
    @ElementCollection
    private List<Long> replies;
}

File: angle-chan-server/src/main/kotlin/com/ben/dektor/entities/Post.kt
package com.ben.dektor.entities


import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonIdentityInfo
import com.fasterxml.jackson.annotation.JsonManagedReference
import com.fasterxml.jackson.annotation.ObjectIdGenerators
import org.hibernate.annotations.CreationTimestamp
import java.sql.Date
import javax.persistence.*
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator::class, property = "number")
@Entity
data class Post (
    @Id
    @GeneratedValue
    var number: Long?,
    var name: String,

    @ManyToOne(cascade = [CascadeType.DETACH])
    var thread: CatalogThread?,

    @CreationTimestamp
    var utc: Date?,
    var text: String,

    @OneToMany(cascade = [CascadeType.DETACH])
    var media: MutableList<MediaContainer>,

    @ElementCollection
    var replies: List<Long>?,
    var isOp: Boolean
    ) {
    constructor(old: Post): this(
            null,
            old.name,
            old.thread,
            old.utc,
            old.text,
            old.media,
            old.replies,
            old.isOp
    )
    constructor(old: Post, isOp: Boolean): this(
            null,
            old.name,
            old.thread,
            old.utc,
            old.text,
            old.media,
            old.replies,
            isOp
    )

}